src/shortestpaths/yen.jl

```diff
@@ -0,0 +1,102 @@
+"""
+    struct YenState{T, U}
+Designed for yen k-shortest-paths calculations.
+"""
+struct YenState{T, U<:Integer}<: AbstractPathState
+    dists::Vector{T}
+    paths::Vector{Vector{U}}
+end
+
+
+"""
+  yen_k_shortest_paths(g, source, target, distmx=DefaultDistance(), K);
+
+  Perform [Yen's algorithm](http://en.wikipedia.org/wiki/Yen%27s_algorithm)
+  on a graph, computing k-shortest distances between `source` and `target` other vertices.
+  Return a [`YenState`](@ref) that contains distances and paths.
+
+  ### Optional Arguments
+  - `k=1`: The number of paths to find.
```

Comment:
upper case `k`? also, you don't need this section if you've defined the default in the signature (see prev. comment). If you decide to keep it, `distmx` is also an optional argument.